src/explain/registry.rs

```diff
@@ -3,7 +3,7 @@ use std::cell::RefCell;
 use std::rc::Rc;
 
 #[derive(Clone, Default, Debug)]
-pub(crate) struct ProofRegistry(Rc<RefCell<HashMap<Equation, ProvenEq>>>);
+pub struct ProofRegistry(Rc<RefCell<HashMap<Equation, ProvenEq>>>);
```

Comment:
This is not something we want to expose to the user, for that reason it's `pub(crate)`.
I guess you get some `unused` warnings again? We might be able to put this struct under the explanations feature flag.

Reply:
Changed visibility of the respective `check` methods to use `pub(crate)` instead.